tests/test_json.py

```diff
@@ -121,10 +121,77 @@ class Squid(proto.Message):
     )
     assert json1 == '{"name":"Steve"}'
 
-    json2 = Squid.to_json(s).replace(" ", "").replace("\n", "")
-    assert (
-        json2 == '{"name":"Steve","massKg":0}' or json2 == '{"massKg":0,"name":"Steve"}'
+    json1 = (
+        Squid.to_json(s, always_print_fields_with_no_presence=False)
+        .replace(" ", "")
+        .replace("\n", "")
     )
+    assert json1 == '{"name":"Steve"}'
+
+    json1 = (
+        Squid.to_json(
+            s,
+            including_default_value_fields=False,
+            always_print_fields_with_no_presence=False,
+        )
+        .replace(" ", "")
+        .replace("\n", "")
+    )
+    assert json1 == '{"name":"Steve"}'
+
+    with pytest.raises(
+        ValueError,
+        match="Arguments.*always_print_fields_with_no_presence.*including_default_value_fields.*must match",
+    ):
+        Squid.to_json(
+            s,
+            including_default_value_fields=True,
+            always_print_fields_with_no_presence=False,
+        ).replace(" ", "").replace("\n", "")
+
+    with pytest.raises(
+        ValueError,
+        match="Arguments.*always_print_fields_with_no_presence.*including_default_value_fields.*must match",
+    ):
+        Squid.to_json(
+            s,
+            including_default_value_fields=False,
+            always_print_fields_with_no_presence=True,
+        ).replace(" ", "").replace("\n", "")
+
+    json2 = (
+        Squid.to_json(
+            s,
+            including_default_value_fields=True,
+            always_print_fields_with_no_presence=True,
+        )
+        .replace(" ", "")
+        .replace("\n", "")
+    )
+    assert json2 == '{"name":"Steve","massKg":0}'
+
+    json2 = (
+        Squid.to_json(
+            s,
+            including_default_value_fields=True,
+        )
+        .replace(" ", "")
+        .replace("\n", "")
+    )
+    assert json2 == '{"name":"Steve","massKg":0}'
+
+    json2 = (
+        Squid.to_json(
+            s,
+            always_print_fields_with_no_presence=True,
+        )
+        .replace(" ", "")
+        .replace("\n", "")
+    )
+    assert json2 == '{"name":"Steve","massKg":0}'
+
+    json2 = Squid.to_json(s).replace(" ", "").replace("\n", "")
```

Comment:
+1 to this test case. It just occurred to me as well to test what happens when we exclude both parameters.